lib/report_plugins/autotune.js

```diff
@@ -0,0 +1,333 @@
+    record.startDate = new Date(splittedString[0]);
+    sortedProfiles.push(record);
+  }
+  sortedProfiles.sort(function (r1, r2) {
+    return (r1.startDate < r2.startDate) ? -1 : ((r1.startDate > r2.startDate) ? 1 : 0);
+  });
+  for (i = 0; i < sortedProfiles.length; i++) {
+    var profile = sortedProfiles[i];
+    if (profile.startDate <= dateFrom) {
+      result.activeIndex = profile.profileIndex;
+      continue;
+    }
+    if (profile.startDate >= dateFrom && profile.startDate <= dateTo)
+      result.manyActive = true;
+    if (profile.startDate > dateFrom)
+      break;
+  }
+
+  return result;
+}
+
+function getHourMinuteText(hour, minute)
+{
+  var text = "";
+  if (hour < 10)
+    text += "0";
+  text += hour.toString();
+  text += ":";
+  if (minute < 10)
+    text += "0";
+  text += minute.toString();
+  return text;
+}
+
+function findBasalRecord(data, time)
+{
+  for(var i = 0; i < data.length; i++)
+  {
+    if (data[i].time == time)
+      return data[i];
+  }
+  return null;
+}
+
+function findRecommendedRecord(data, minutes) {
+  for(var i = 0; i < data.length; i++)
+  {
+    if (data[i].minutes == minutes)
+      return data[i];
+  }
+  return null;
+}
+
+autotune.report = function report_autotune(datastorage, sorteddaystoshow, options) {
+  var Nightscout = window.Nightscout;
+  var client = Nightscout.client;
+  var translate = client.translate;
+  var report_plugins = Nightscout.report_plugins;
+
+  if (!$('#compute_autotune').is(':checked')) return;
+
+  var profileRecords = datastorage.profiles;
+
+  var report = $('#autotune-report');
+  report.empty();
+
+  var foundActiveProfile = findActiveProfiles(datastorage.profiles, sorteddaystoshow);
+  if (foundActiveProfile.activeIndex < 0 || foundActiveProfile.manyActive)
+  {
+    // Show error
+    $('<div class="error">' + "Error: Many profiles are active during choosen period." + "</div>").appendTo(report);
+    return;
+  }
+
+  var table = $('<table class="centeraligned autotune-report">');
+  report.append(table);
+
+  var thead = $('<tr/>');
+  $('<th/>').appendTo(thead);
+  $('<th>' + translate('Current NS Profile') + '</th>').appendTo(thead);
+  $('<th>' + translate('Calculated') + '</th>').appendTo(thead);
+  $('<th/>').appendTo(thead);
+  thead.appendTo(table);
+
+  var profile = profileRecords[foundActiveProfile.activeIndex];
+  var store = profile.store[profile.defaultProfile];
+
+  const re = /\d{4}-\d{2}-\d{2}/g;
+  var keys = _.filter(Object.keys(datastorage), function (key) { return key.match(re)});
+
+  var convertedProfile = autotune.convertProfile(profile);
+  var opts = {
+        profile: convertedProfile //inputs.profile
+      // TODO: figure out how to use only the relevant treatments so it's not so slow on >1d of data
+      , history: _.flatMap(keys, function(key){return datastorage[key].treatments;}) //inputs.history
+      , glucose: _.flatMap(keys, function(key){return datastorage[key].sgv;})//inputs.glucose
```

Comment:
To be confirmed, but I think datastorage[key].sgv is giving glucose in mmol, rather than mg/dl for mmol users.

Reply:
All the stored sgv values _should_ be in mg/dl in Nightscout - if someone has implemented an uploader that sends mmol values to NS, all bets are off with how the software works. Can you confirm what the data format is for you?

Reply:
Something in the reporting layer must be scaling it back to mmol i think. I saw the values as mmol under the debugger, i'll try again to see where the scaling is done.

Reply:
seems like this line should be changed to:
`     , glucose: _.flatMap(keys, function(key){return datastorage[key].y;})//inputs.glucose`
scaling is being done here, but `y` has the unscaled value.
https://github.com/nightscout/cgm-remote-monitor/blob/master/static/report/js/report.js#L767

Reply:
ok, i got this slightly wrong. This line of code is correct, i'll add another comment where i needed to change to use y to make this work.

Reply:
Just added mmol -> mgdl conversion to work around this